indexer/services/roundtable/__tests__/tasks/update-affiliate-info.test.ts

```diff
@@ -0,0 +1,276 @@
+      FillTable.create({
+        ...testConstants.defaultFill,
+        liquidity: Liquidity.TAKER,
+        createdAt: DateTime.utc().toISO(),
+        eventId: testConstants.defaultTendermintEventId,
+        price: '1',
+        size: '1',
+        fee: '1000',
+        affiliateRevShare: '500',
+      }),
+      AffiliateReferredUsersTable.create({
+        affiliateAddress: testConstants.defaultWallet2.address,
+        refereeAddress: testConstants.defaultWallet3.address,
+        referredAtBlock: '2',
+      }),
+    ]);
+
+    updatedDt = DateTime.utc();
+    await BlockTable.create({
+      blockHeight: '4',
+      time: updatedDt.toISO(),
+    });
+
+    await affiliateInfoUpdateTask();
+
+    updatedInfo = await AffiliateInfoTable.findById(
+      testConstants.defaultWallet2.address,
+    );
+    expectedAffiliateInfo = {
+      address: testConstants.defaultWallet2.address,
+      affiliateEarnings: '500',
+      referredMakerTrades: 0,
+      referredTakerTrades: 1,
+      totalReferredFees: '1000',
+      totalReferredUsers: 2,
+      referredNetProtocolEarnings: '500',
+      firstReferralBlockHeight: '1',
+      referredTotalVolume: '1',
+    };
+    expect(updatedInfo).toEqual(expectedAffiliateInfo);
+    lastUpdateTime = await getAffiliateInfoUpdateTime();
+    if (lastUpdateTime !== undefined) {
+      expect(lastUpdateTime.toMillis()).toEqual(updatedDt.toMillis());
+    }
+  });
+
+  it('Successfully backfills from past date', async () => {
+    const currentDt: DateTime = DateTime.utc();
+
+    // Set persistent cache to 3 weeks ago to emulate backfill from 3 weeks.
+    await PersistentCacheTable.create({
+      key: PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME,
+      value: currentDt.minus({ weeks: 3 }).toISO(),
+    });
+
+    // defaultWallet2 will be affiliate and defaultWallet will be referee
+    await AffiliateReferredUsersTable.create({
+      affiliateAddress: testConstants.defaultWallet2.address,
+      refereeAddress: testConstants.defaultWallet.address,
+      referredAtBlock: '1',
+    });
+
+    // Fills spannings 2 weeks
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      liquidity: Liquidity.TAKER,
+      createdAt: currentDt.minus({ weeks: 1 }).toISO(),
+      eventId: testConstants.defaultTendermintEventId,
+      price: '1',
+      size: '1',
+      fee: '1000',
+      affiliateRevShare: '500',
+    });
+    await FillTable.create({
+      ...testConstants.defaultFill,
+      liquidity: Liquidity.TAKER,
+      createdAt: currentDt.minus({ weeks: 2 }).toISO(),
+      eventId: testConstants.defaultTendermintEventId2,
+      price: '1',
+      size: '1',
+      fee: '1000',
+      affiliateRevShare: '500',
+    });
+
+    // Create block at current time
+    await BlockTable.create({
+      blockHeight: '3',
+      time: DateTime.utc().toISO(),
+    });
+
+    // Simulate backfill
+    let backfillTime: DateTime | undefined = await getAffiliateInfoUpdateTime();
```

Comment:
same feedback as above with the let

Reply:
But with this one i actually want to update the value as part of the while loop control.